openmdao/surrogate_models/kriging.py

```diff
@@ -150,6 +157,19 @@ def _calculate_reduced_likelihood_params(self, thetas=None):
         params['U'] = U
         params['Vh'] = Vh
 
+        # Newly added for EGOLF
+        one = np.ones([self.n_samples,1])
+        R_inv = Vh.T.dot(np.dot(np.diag(inv_factors),U.T))
```

Comment:
Using `einsum` rather than `np.diag` will be much more performant and use less memory.